common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1007,6 +1007,32 @@ public AsciiString toUpperCase() {
 
     /**
      * Copies this string removing white space characters from the beginning and end of the string.
+     * @param c The {@link CharSequence} to trim.
+     * @return a new string with characters {@code <= \\u0020} removed from the beginning and the end.
+     */
+    public static CharSequence trim(CharSequence c) {
+        if (c.getClass() == AsciiString.class) {
```

Comment:
why we not use an instanceof here as well ?

Reply:
we use a direct class check because `AsciiString` is a `final` class. This is consistent with other checks in this class.